src/components/Authentication/OAuth.vue

```diff
@@ -0,0 +1,53 @@
+<template lang="pug">
+v-container.oauth-widget
+  h4 Or {{ verb }} with OAuth
+  v-btn.ml-0.mr-3(v-for="provider in providers",
+      :key="provider.id",
+      :dark="iconMap[provider.id].dark",
+      :color="provider.id",
+      :href="provider.url")
+    v-icon.aligned(left) {{ $vuetify.icons[iconMap[provider.id].icon] }}
+    | {{ provider.name }}
+</template>
+
+<script>
+const iconMap = {
+  github: { dark: true, icon: 'github' },
+  google: { dark: true, icon: 'google' },
+  linkedin: { dark: true, icon: 'linkedin' },
+  bitbucket: { dark: false, icon: 'bitbucket' },
+  box: { dark: true, icon: 'box_com' },
+  globus: { dark: true, icon: 'globus' },
+};
+export default {
+  props: {
+    verb: {
+      type: String,
+      default: 'sign in',
+    },
+    // Array of Object{ id: String, name: String, url: String }
+    providers: {
+      type: Array,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      enabledProviders: [],
+      iconMap,
+    };
+  },
+};
+</script>
+
+<style lang="stylus" scoped>
+.v-btn
+  &.google
+    background-color #3367d6 !important
```

Comment:
Sorry for the run-around on this particular section of code -- I re-read it and realized we are doing something that may be relying on undefined behavior. You are passing `:color="provider.id"` to the button, which does add this class, but that may be an implementation detail. I think it would be better to remove this style block altogether, add the colors as a new field into the `iconMap`, and pass `:color="iconMap[provider.id].color"`.

Aside for general proselytizing: @jeffbaumes has ingrained the idea in me that we should have as little code in `style` blocks as possible, as they (often) represent deviations from the norms of our styling framework.